### ISSUE TITLE
Course start date is shown one day earlier in public area as entered in login area

### ISSUE THREAD

**Issue by PixlTracer** [acdh-oeaw/dhcr-main#113]
Issue reported by @LauraEst:

I’ve added a couple of programs to the DH Course Registry [...].

Even though I’ve entered the dates correctly:
<img width="1021" alt="Screenshot_Dates_Backend" src="https://github.com/user-attachments/assets/607cb86e-e05f-4be4-9c83-52e60d729ec5">

You can see that the calendar has the dates all one day earlier than the ones I’ve input:
<img width="698" alt="Screenshot_Dates_Frontend" src="https://github.com/user-attachments/assets/043332a3-5667-4d46-9495-cc907309d6c8">

This has happened for all three that I entered. Should I leave it as is?
https://dhcr.clarin-dariah.eu/courses/view/848
https://dhcr.clarin-dariah.eu/courses/view/850
https://dhcr.clarin-dariah.eu/courses/view/849

Comment @PixlTracer:
my guess would be that this issue occurs because @LauraEst is located in a different time zone (ADT vs. CET), but I could be wrong...
@patrickakk Perhaps this is related to Laura's time zone (Antigonish, Canada, Nova Scotia), could this be amended if she would simply set the start date to one day later than indicated?

**Comment by patrickakk** [comment]
@LauraEst @PixlTracer 

Thanks for the report.

For now, I can give the following update:
I've checked what was stored in the database. The dates are stored correctly, exactly as shown on the first screenshot from the login area:

`2024-12-01;2025-01-15;2025-04-01`

The second screenshot shows the public area, at a course detail page. At this point the data is processed by JavaScript to show it to the user. This is where it seems to go wrong. As @PixlTracer also suggested, this might have to do with a different timezone. This probably should not happen. It would be nice if the application behaves the same in other time zones than GMT +1 ? 

That needs some further investigation.

**Comment by patrickakk** [comment]
@LauraEst @PixlTracer
A possible fix has been implemented. I'll keep you updated when it's available in test, so we could test it in Laura's timezone.

**Comment by patrickakk** [comment]
`Note: synced prod and test DB now.`

**Comment by LauraEst** [comment]
update: this seems to be browser specific. (I deleted cookies to test to make sure that my history wasn't affecting anything; the only browser I used in the past to access the site was Chrome.)

Tor Browser: Correct Dates
<img width="783" alt="Screenshot 2024-11-04 at 3 13 19 PM" src="https://github.com/user-attachments/assets/894e61c8-d629-4ac3-b887-a39a69d202f5">

Chrome Browser: Incorrect Dates
<img width="712" alt="Screenshot 2024-11-04 at 3 14 14 PM" src="https://github.com/user-attachments/assets/68b896cf-23a0-4d88-a5f7-f252b0512056">

Safari Browser: Incorrect Dates
<img width="808" alt="Screenshot 2024-11-04 at 3 14 49 PM" src="https://github.com/user-attachments/assets/c631e9b8-efed-4d3d-adaa-2ac2b30fd178">

URL: https://dhcr.clarin-dariah.eu/courses/view/848

**Comment by patrickakk** [comment]
@LauraEst @PixlTracer @IvdL22 
The fix is now available in test:
https://test-dhcr.clarin-dariah.eu/

Can you review this?

@LauraEst 
Thanks for testing it with multiple browsers and this interesting result. Maybe the TOR browser uses a different location and therefore a different timezone?

Do you want to participate in testing the fix? 
If so, I could send you the credentials per mail.
The url is specified above. Please note that the test instance uses a separate database, which might contain different data then the public version.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 

You probably need to use a different timezone to test this.

**Comment by PixlTracer** [comment]
hi @patrickakk 
I don't know how to test this in a different time zone other than the one I am operating in.
could you provide a setp-by-step guide on how to test this in a different time zone?
(I want to avoid to change the date/time settings on the device I am working on.)

**Comment by IvdL22** [comment]
@PixlTracer @patrickakk Maybe it is easier if we ask some National Moderators who live in different timezones to test this?

**Comment by LauraEst** [comment]
<img width="707" alt="Screenshot 2024-11-13 at 2 48 01 PM" src="https://github.com/user-attachments/assets/3fef6801-7339-4ffd-a6a1-301303ca5e87">

I'm still getting incorrect dates in Chrome Browser and Safari

Sorry for the delay on this--I was travelling last week

**Comment by patrickakk** [comment]
@LauraEst 
Thank you for confirming that you would like to help to test it. I'll send you the additional credentials for the test instance per mail.

On this project the new features/changes are tested before they are made available in the public available version of the application (production instance).

For this purpose, we have a test instance available. Only after testing/reviewing and confirming that everyhing works as intended/requested, the changes will appear in production.

You can find the test instance here:
https://test-dhcr.clarin-dariah.eu/

The detail pages for the courses you specified can be found here:
https://test-dhcr.clarin-dariah.eu/courses/view/848
https://test-dhcr.clarin-dariah.eu/courses/view/849
https://test-dhcr.clarin-dariah.eu/courses/view/850


Could you confirm if the dates are shown correct, using any browser, except TOR ?

Please note that the test instance uses a *separate database*. Changes made there do not affect production.

@IvdL22 @PixlTracer 
My suggestion would be to wait briefly if @LauraEst could test this. That might be the most efficient way. And what is better than when the person who reports a bug, can also confirm that everything works fine?

**Comment by LauraEst** [comment]
Just checked on the test site (using Chrome) and it look goods to me! 
Thank you!

**Comment by patrickakk** [comment]
@LauraEst Thank you for your help. It's nice to read that everything works fine now.

Version 2024-11 has been released into production today, since the review process of other features also was finished. This means that the fix is now also available in production.

Would it be possible for you to add a screenshot that shows the url and the correct dates? Then we have everything documented, since this problem only appeared in your timezone. Maybe from this url:
https://test-dhcr.clarin-dariah.eu/courses/view/848

In case you find another problem in the future, feel free to create a new issue here:
https://github.com/acdh-oeaw/dhcr-main/issues/new/choose

or send an email to Anna ;)

@PixlTracer @IvdL22 
Can you change the label to done, if you agree with the solution?


Status update:
Implemented in 2024-11

**Comment by IvdL22** [comment]
@patrickakk Thank you for this. Shall I mark it now as Done or do you want the screenshot from Laura?

**Comment by patrickakk** [comment]
@IvdL22 You could mark it as done when you also agree?
I'll wait for the screenshot from Laura, before closing the issue.

**Comment by LauraEst** [comment]
<img width="1051" alt="Screenshot 2024-11-15 at 4 08 39 PM" src="https://github.com/user-attachments/assets/e94373b1-2847-415c-9757-d8eb6cfb6bcc">
Here's a screenshot showing the correct dates from the test site

**Comment by IvdL22** [comment]
@patrickakk I will mark this as done.

**Comment by patrickakk** [comment]
@LauraEst Thank you for your help and adding the screenshot.
